packages/dd-trace/test/appsec/reporter.spec.js

```diff
@@ -141,17 +142,19 @@ describe('reporter', () => {
     })
 
     it('should set duration metrics if set', () => {
-      Reporter.reportMetrics({ duration: 1337 })
+      const metrics = { duration: 1337 }
+      Reporter.reportMetrics(metrics)
 
       expect(web.root).to.have.been.calledOnceWithExactly(req)
-      expect(span.setTag).to.have.been.calledOnceWithExactly('_dd.appsec.waf.duration', 1337)
+      expect(telemetry.updateWafRequestsMetricTags).to.have.been.calledOnceWithExactly(metrics, req)
```

Comment:
I'm not always a fan of testing with the source object instead of an hardcoded value.
If you're trying to test that the exact reference of that object is passed, then use the object like that.
But if you're trying to test if the value inside the object is passed, I think it's better to hardcode the value. Otherwise you're not testing the case where that object gets mutated.

Reply:
I want to test that `telemetry.updateWafRequestsMetricTags` is called. The `metrics` argument should be the exact reference.